operator/v1/types_cloudcredential.go

```diff
@@ -0,0 +1,63 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudCredential provides a means to configure an operator to manage CredentialsRequests.
+type CloudCredential struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// +required
+	Spec CloudCredentialSpec `json:"spec,omitempty"`
+	// +optional
+	Status CloudCredentialStatus `json:"status,omitempty"`
+}
+
+// CloudCredentialsMode is the specified mode the cloud-credential-operator
+// should reconcile CredentialsRequest with
+// +kubebuilder:validation:Enum="";manual;mint;passthrough
+type CloudCredentialsMode string
+
+const (
+	// CloudCredentialsModeManual tells cloud-credential-operator to not reconcile any CredentialsRequests
+	// (primarily used for the disconnected VPC use-cases).
+	CloudCredentialsModeManual CloudCredentialsMode = "manual"
+
+	// CloudCredentialsModeMint tells cloud-credential-operator to reconcile all CredentialsRequests
+	// by minting new users/credentials.
+	CloudCredentialsModeMint CloudCredentialsMode = "mint"
+
+	// CloudCredentialsModePassthrough tells cloud-credential-operator to reconcile all CredentialsRequests
+	// by copying the cloud-specific secret data.
+	CloudCredentialsModePassthrough CloudCredentialsMode = "passthrough"
+)
+
+// CloudCredentialSpec is the specification of the desired behavior of the cloud-credential-operator.
+type CloudCredentialSpec struct {
+	OperatorSpec `json:",inline"`
+	// ForceCredentialsMode allows informing CCO that it should not attempt to dynamically
+	// determine the root cloud credentials capabilities, and it should just run in
+	// the specified mode.
+	ForceCredentialsMode CloudCredentialsMode `json:"forceCredentialsMode,omitempty"`
```

Comment:
Do we want to take the same stance as the install-config and rename this `CredentialsMode`? See https://github.com/openshift/installer/pull/3919#discussion_r456729954 from the installer PR. I think it applies here as well.

Reply:
As you pointed out, it is highly recommended to put the +optional https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required.